deploy/lib/xquery/setup.xqy

```diff
@@ -3136,6 +3140,59 @@ declare function setup:validate-group(
       setup:validation-fail(fn:concat("Missing Group: ", $group))
 };
 
+declare function setup:validate-group-settings(
+  $group-config as element(gr:group),
+  $setting as element()
+) {
+  let $group := $group-config/gr:group-name/fn:string(.)
+  let $setting-name := $setting/fn:string(.)
+  let $function-name := fn:concat("group-get-", $setting-name, "s")
+  let $keys := fn:tokenize($setting/@keys, " ")
+  let $existing :=
+    try
+    {
+      xdmp:eval(
```

Comment:
Look like the only reason you are evaling here is to return the function. Instead, couldn't you do this?:

```
let $config := admin:get-configuration()
let $groupid := admin:group-get-id($config, $group)
return xdmp:value(fn:concat('admin:', $function-name, '($config, $groupid)'))
```

Granted, I only played around with this a little in QConsole so let me know if I'm overlooking something here...